src/handlers/handleIngressAPI.ts

```diff
@@ -22,7 +30,34 @@ async function makeIngressRequest(receivedRequest: Request, env: IntegrationEnv)
   addProxyIntegrationHeaders(request.headers, receivedRequest.url, env)
 
   console.log(`sending ingress request to ${url.toString()}...`)
-  return fetch(request, { backend: 'fpjs' })
+
+  if (!isOpenClientResponseEnabled(env)) {
+    return fetch(request, { backend: 'fpjs' })
+  }
+
+  const secretStore = new SecretStore(FingerprintSecretStoreName)
+  const decryptionKey = await secretStore.get(FingerprintDecryptionKeyName).then((v) => v?.plaintext())
+  if (!decryptionKey) {
+    throw new Error('Decryption key not found in secret store')
+  }
+
+  try {
+    const response = await fetch(request, { backend: 'fpjs' })
+
+    // Parse the open response
+    const text = await response.text()
+    const event = await unsealData(JSON.parse(text).sealedResult, decryptionKey)
+    void processOpenClientResponse({ event, sealedResult: JSON.parse(text).sealedResult }) // void means skip awaiting this and handle it in the background
+
+    return new Response(text, {
+      headers: response.headers,
+      status: response.status,
+      statusText: response.statusText,
+    })
+  } catch (e) {
+    console.error('ingress request failed', e)
```

Comment:
we are throwing error if `unsealData` or `JSON.parse` fails. I think these should be separated. If proxy fails to unseal the data, it should still continue to return the response to the agent, with optional logging of what went wrong. Not being able to unseal the data should not get in the way of HTTP response delivery to the JS agent IMO. What do you think?

Reply:
As we all discussed this, I changed it as "return response itself if unseal fails".
Changed by this commit: https://github.com/fingerprintjs/fingerprint-pro-fastly-proxy-integration-example/pull/5/commits/baba5e708fcc502e6a1665a6e6e53ef9a1677e6f